core-web/libs/template-builder/src/lib/components/template-builder/utils/gridstack-utils.ts

```diff
@@ -189,9 +191,37 @@ export const parseFromGridStackToDotObject = (gridData: DotGridStackWidget[]): D
  * @return {*}  {TemplateBuilderBoxSize}
  */
 export function getBoxVariantByWidth(width: number): TemplateBuilderBoxSize {
-    if (width <= 1) return TemplateBuilderBoxSize.small;
+    if (width <= 1) {
+        return TemplateBuilderBoxSize.small;
+    }
 
-    if (width <= 3) return TemplateBuilderBoxSize.medium;
+    if (width <= 3) {
+        return TemplateBuilderBoxSize.medium;
+    }
 
     return TemplateBuilderBoxSize.large;
 }
+
+/**
+ * @description This function returns if a box will fit in a row
+ *
+ * @export
+ * @param {DotGridStackWidget} row
+ * @return {*} boolean
+ */
+export function willBoxFitInRow(row: DotGridStackWidget): boolean {
+    const emptyChar = '_';
+    const boxChar = '#';
+
+    if (!row.subGridOpts) {
+        return false;
+    }
+
+    const rowSpace = Array.from({ length: 12 }, () => emptyChar); // Array with 12 empty spaces
+
+    row.subGridOpts.children.forEach(({ x, w }) => {
+        rowSpace.splice(x, w, ...boxChar.repeat(w).split('')); // Fill needed empty spaces with #
+    });
+
+    return /___/.test(rowSpace.join('')); // If there are 3 consecutive empty spaces then we can drop one more box
```

Comment:
If we change the width of the new box we will have to remember that we need to change it here... can we create somehow a constant that we can just change in one place?